### PR TITLE
[record-minmax] Separate RecordFunction.h

### PR DESCRIPTION
**Pull request by jinevening**
This separates RecordFunction.h to .h and .cpp.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/pull/11410#issuecomment-1700518870

### CODE REVIEW

**Comment by jinevening** [comment]
## Motivation

This separates RecordFunction.h into .h and .cpp, which allows to include RecordFunction.h multiple times by other sources.

File: compiler/record-minmax/include/RecordFunction.h [hunk 1]
```diff
@@ -14,12 +14,11 @@
  * limitations under the License.
  */
 
+#ifndef __RECORD_MINMAX_RECORD_FUNCTION_H__
+#define __RECORD_MINMAX_RECORD_FUNCTION_H__
```
Comment (jinevening) [inline compiler/record-minmax/include/RecordFunction.h:18]: I've added header guard, so that this header can be used by multiple sources.

**Review by seanshpark — DISMISSED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by glistening** [comment]
@jinevening Just out of curiosity, `RecordFunction.h` seems not to record. It gets `NthPercentile` and `MovingAvg`. Why it is named as `RecordFunction`? Historical reason? Or because it is for `record-minmax`?

**Comment by jinevening** [comment]
> Just out of curiosity, RecordFunction.h seems not to record. It gets NthPercentile and MovingAvg. Why it is named as RecordFunction? Historical reason? Or because it is for record-minmax?

Ah, it is just because for record-minmax. We can use a better name!

**Review by glistening — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by glistening — APPROVED** [review]
LGTM

**Pull request merged**